src/common/TrackedOp.cc

```diff
@@ -52,17 +50,11 @@ void OpHistory::cleanup(utime_t now)
   while (arrived.size() &&
 	 (now - arrived.begin()->first >
 	  (double)(history_duration))) {
-    duration.erase(make_pair(
-	arrived.begin()->second->get_duration(),
-	arrived.begin()->second));
-    arrived.erase(arrived.begin());
+     arrived.erase(arrived.begin());
   }
 
-  while (duration.size() > history_size) {
-    arrived.erase(make_pair(
-	duration.begin()->second->get_initiated(),
-	duration.begin()->second));
-    duration.erase(duration.begin());
```

Comment:
Here we're trimming based on whatever sort order the duration set is using. I think the intention here is to delete the ops which took the least time to process, but I don't know if that's what actually happens.

If we want to make a change, I think the only appropriate one would be here, to make sure we're deleting the least-interesting ops (those of shortest duration).